interceptors/realip/realip.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) The go-grpc-middleware Authors.
+// Licensed under the Apache License 2.0.
+
+package realip
+
+import (
+	"context"
+	"net"
+	"net/netip"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+)
+
+// X_REAL_IP, X_FORWARDED_FOR and TRUE_CLIENT_IP are header keys
+// used to extract the real client IP from the request. They represent common
+// conventions for identifying the originating IP address of a client connecting
+// through proxies or load balancers.
+const (
+	X_REAL_IP       = "x-real-ip"
+	X_FORWARDED_FOR = "x-forwarded-for"
+	TRUE_CLIENT_IP  = "true-client-ip"
```

Comment:
Absolutely - oversight on my part.

Reply:
Sorry to bang on, but I don't see that the case of the headers was canonicalized?

Reply:
Hey @johanbrandhorst - I guess I must have misunderstood the meaning of canonicalising the header keys. Let me go ahead and try to fix them ;)

Reply:
I've changed the casing of the header keys that are the values of the constants - Is that what you meant by canonicalising the header keys?

Reply:
Yes, that is what I meant. I see you had to lowercase the headers in the map, which is fine, you could also canonicalize them using https://pkg.go.dev/net/textproto#CanonicalMIMEHeaderKey, but this will do.